dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1287,10 +1311,18 @@ private void registerZenodoDOIForWorkflow(String zenodoAccessToken, Workflow wor
             }
         }
 
-        provisionWorkflowVersionUploadFiles(zendoClient, returnDeposit,
-                depositionID, workflow, workflowVersion);
 
-        putDepositionOnZenodo(depositApi, depositMetadata, depositionID);
+        File uploadFile = provisionWorkflowVersionUploadFiles(zendoClient, returnDeposit, depositionID, workflow, workflowVersion);
```

Comment:
This could use a comment... looking at code It seems odd that you have to wait for line 1317 to finish executing before you can delete the file, even though you don't pass the file to that method. As you explained to me some part of the deposition has a reference to the file so it can't be deleted, but if I saw this code 1 year from now I would be tempted to refactor it.

Reply:
https://ucsc-cgl.atlassian.net/browse/DOCK-860